### ISSUE TITLE
item.comments.add( "New comment" ) causes 'set' on proxy: trap returned falsish for property 'replies'

### ISSUE THREAD

**Issue by DennisKuhn** [pnp/pnpjs#1834]
### Category
- [ ] Enhancement
- [x] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [2.7.0]

Please specify what version(s) of SharePoint you are targeting: [Online]

### Expected / Desired Behavior / Question
Call `await listItem.comments.addComment( "new comment")` and retrieve the new comment.

### Observed Behavior
The comment is successful added to the SharePoint List item. Then the following [return assign(this.getById(d.id), d);](https://github.com/pnp/pnpjs/blob/88d92129b7e7310447374ffcf3d77be4c5fdc436/packages/sp/comments/types.ts#L33)
causes an an exception:
```
message: 'set' on proxy: trap returned falsish for property 'replies'
TypeError: 'set' on proxy: trap returned falsish for property 'replies'
    at https://localhost:4321/dist/announcements.bundle.js:332726:14
    at Array.reduce (<anonymous>)
    at assign (https://localhost:4321/dist/announcements.bundle.js:332725:10)
    at Proxy.<anonymous> (https://localhost:4321/dist/announcements.bundle.js:108210:105)
    at step (https://localhost:4321/dist/announcements.bundle.js:299541:23)
    at Object.next (https://localhost:4321/dist/announcements.bundle.js:299522:53)
    at fulfilled (https://localhost:4321/dist/announcements.bundle.js:299512:58)
```

### Steps to Reproduce
```
            const pnpItem = this.list.items.getById(id);
            try {
                const newComment = await pnpItem.comments.add(`new comment`);
                console.log(`Never gets here`, {newComment, pnpItem});
            } catch( commentError ) {
                console.error(`Caught adding Comment`, {commentError, pnpItem});
            }
```

**Comment by patrick-rodgers** [comment]
Thanks, will have a look.

**Comment by patrick-rodgers** [comment]
I am not able to reproduce this issue. My code:

```
const item = await sp.web.lists.getByTitle("CommentList").items.getById(1).comments.add("test");
```

In what browser are you encountering this issue?

**Comment by hafeez1042** [comment]
Having similar issue on edge browser

**Comment by DennisKuhn** [comment]
TLDR: don't use `odata=verbose`
When using the header setting `odata=verbose`, the response contains for the property `replies` an `{ __deferred: { ...  }}`. I was using isolated context, it works now with:
```
        const isolatedSp = await sp.createIsolated({
            baseUrl: normalisedSiteUrl
        });

        isolatedSp.setup({
            sp: {
                baseUrl: normalisedSiteUrl,
                headers: {
                    // --> Causes { __deferred: } for empty arrays: "Accept": "application/json;odata=verbose;charset=utf-8"
                    "Accept": "application/json;charset=utf-8"
                }
            }
        });
```

**Comment by patrick-rodgers** [comment]
Glad you figured it out, we would never have from the description provided. Verbose mode isn't something we really test as it has a lot of weird side effects. Looks like you ran into one. Thanks for updating the ticket and using PnPjs!
